app/src/main/java/net/gsantner/markor/util/AppSettings.java

```diff
@@ -763,12 +782,10 @@ public String getHuuidDeviceId() {
         return deviceid;
     }
 
-    public boolean hasPasswordBeenSetOnce() {
-        return getBool(R.string.pref_key__default_encryption_password_set_once, false);
-    }
-
-    public void setPasswordHasBeenSetOnce(boolean b) {
-        setBool(R.string.pref_key__default_encryption_password_set_once, b);
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    public String getPassword() {
```

Comment:
Would be better to use char[] here. Lesser conversion and lesser String-Caching.

Reply:
To be honest, a char[] to string and vice versa, of lets say 64 chars max .... is not really performance critical.  
operations on the user file (text content) is important.

Whatever is easier working for the use here [[ less code ]] , is better.

Reply:
string vs char[] is not an performance issue, it's an security issue - at least at normal java runtimes.